homework.js

```diff
@@ -0,0 +1,108 @@
+//  1.
+//  sum all numbers, from 1 to num
+const summation = function (num) {
+    let sum = 0;
+    for (i = 1; i <= num; i++) {
+        sum += i;
+    }
+    return sum;
+}
+
+//  2.
+//  sum all even numbers equal or less than num
+const summationEven = function (num) {
+    let sum = 0;
+    if (num % 2 == 1)
+        num--;
+
+    return summation(num / 2) * 2;
+}
+
+//  3.
+//  average numbers array
+const avg = function (numArr) {
+    let sum = 0;
+    for (i = 0; i < numArr.length; i++) {
+        sum += numArr[i];
+    }
+    return sum / numArr.length;
+}
+
+//  4.
+//  reverse string
+const reverse = function (inString) {
+    let splitString = inString.split("");
+    let reverseArray = splitString.reverse();
+    return reverseArray.join("");
+}
+
+//  5.
+//  combine an array of words with dash
+const addDashes = function (wordArr) {
+    return wordArr.join("-");
+}
+
+//  6.
+//  count up then down in a string
+const countUpAndDown = function (num) {
+    const countArr = [num];
+    for (i = num - 1; i > 0; i--) {
+        countArr.unshift(i);
+        countArr.push(i);
+    }
+    return countArr.join(" ");
+}
+
+//  7.
+//  list words with an A from a word array
+const wordsWithA = function (wordArr) {
+    const wordsA = [];
+    let wordS = ""
+    for (i = 0; i <= wordArr.length; i++) {
```

Comment:
- Don't forget to initialize the variable `i`.
- the last index in an array is its `length - 1` so when you loop through an array  the condition should looks like this :
`for (let i = 0; i < wordArr.length; i++)`, as you don't need the equal sign.